### PR TITLE
bpo-44676: Serialize the union type using only public API

### PR DESCRIPTION
**Pull request by serhiy-storchaka**
https://bugs.python.org/issue44676

### CODE REVIEW

**Comment by uriyyo** [comment]
Should we also remove `Union._from_args` classmethod introduced by #27244?

File: Lib/copyreg.py [hunk 1]
```diff
@@ -36,6 +36,12 @@ def pickle_complex(c):
 
     pickle(complex, pickle_complex, complex)
 
+def pickle_union(obj):
```
Comment (Fidget-Spinner) [inline Lib/copyreg.py:39]: Off topic: I didn't know such a module existed 😮 . This is really cool! Thanks for teaching me something new today Serhiy.

**Review by Fidget-Spinner — COMMENTED** [review]
> Should we also remove Union._from_args classmethod introduced by #27244? Basically, it was introduced for pickling support so looks like we don't need it anymore.

Agreed. Lets remove it.

Note: we can only remove it if we're backporting this to 3.10. Pickle docs says its confirmed to be [backwards compatible](https://docs.python.org/3/library/pickle.html#:~:text=The%20pickle%20serialization%20format%20is,unique%20breaking%20change%20language%20boundary). So if we don't backport this, we have to leave the private method in forever :(. Either that, or we can remove serialization from 3.10, I created a PR at GH-27335 for that.

If Serhiy agrees to backporting this, we should probably consult the RM Pablo again.

**Comment by Fidget-Spinner** [comment]
Copied over from GH-27244, by Yurii:
> @Fidget-Spinner I just realise that Union._from_args will be removed by #27323. I think we should backport #27323 to not introduce Union._from_args at python 3.10.

I agree with Yurii's reasoning here. +1 to backport this to 3.10 to remove `_from_args` from the API.

**Comment by serhiy-storchaka** [comment]
Oh, I did not know this feature was backported to 3.10. It makes sense to haste and backport this PR too.

**Pull request merged**